### PR TITLE
Update to golang 1.17

### PR DESCRIPTION
**Pull request by JohnStrunk**
**Describe what this PR does**
Updates the version of golang used to build the operator from 1.16 to 1.17

**Is there anything that requires special attention?**

**Related issues:**
openshift/release#23446

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -22,3 +22,96 @@ require (
 	k8s.io/kubectl v0.22.1
 	sigs.k8s.io/controller-runtime v0.10.0
 )
+
+require (
```
Comment (JohnStrunk) [inline go.mod:26]: The require block was added by `go mod tidy`

**Review by tesshuflower — APPROVED** [review]
/lgtm

**Pull request merged**